bytecode.py

```diff
@@ -28,16 +36,90 @@ class BB(ABC):
 class Var(Value):
     name: str
 
+    def lookup_self(self, env: Dict[Var, SExp]) -> SExp:
+        return env[self]
+
 
 @dataclass(frozen=True)
 class NumLit(Value):
     value: SNum
 
+    def lookup_self(self, env: Dict[Var, SExp]) -> SExp:
+        return self.value
+
 
 @dataclass(frozen=True)
 class SymLit(Value):
     value: SSym
 
+    def lookup_self(self, env: Dict[Var, SExp]) -> SExp:
+        return self.value
+
+
+class EvalEnv:
+    _local_env: Dict[Var, SExp]
```

Comment:
`Value` is actually misnamed here, and I named it thinking about LLVM's naming. I should probably rename it to `Parameter` or something like that, since it's intended to be "things that can be used as an operand for an instruction."

My intention is to have it be `Var` -> `SSym` + `SNum` + `SVect`, which are the valid runtime values.

Reply:
So could we use Union[SSym, SNum, SVect] instead of SExp as the mapped-to type?

Reply:
I like `Parameter` over `Value`